jlexer/lexer.go

```diff
@@ -240,23 +242,50 @@ func (r *Lexer) fetchNumber() {
 
 // findStringLen tries to scan into the string literal for ending quote char to determine required size.
 // The size will be exact if no escapes are present and may be inexact if there are escaped chars.
-func findStringLen(data []byte) (isValid, hasEscapes bool, length int) {
-	delta := 0
-
-	for i := 0; i < len(data); i++ {
-		switch data[i] {
-		case '\\':
-			i++
-			delta++
-			if i < len(data) && data[i] == 'u' {
-				delta++
-			}
-		case '"':
-			return true, (delta > 0), (i - delta)
+func findStringLen(data []byte) (isValid bool, length int) {
+	for {
+		idx := bytes.IndexByte(data, '"')
+		if idx == -1 {
+			return false, len(data)
+		}
+		if idx == 0 || (idx > 0 && data[idx-1] != '\\') {
+			return true, length + idx
+		}
+		length += idx + 1
+		data = data[idx+1:]
+	}
+}
+
+// unescapeStringToken performs unescaping of string token.
+// if no escaping is needed, original string is returned, otherwise - a new one allocated
+func (r *Lexer) unescapeStringToken() (err error) {
+	data := r.token.byteValue
+	var unescapedData []byte
```

Comment:
I think that you need to preallocate array with cap = len(data)

Reply:
Hmmm, I can't explain it, but while preallocating makes number of allocations in benchmark lower (52 -> 46), the performance gain becomes less - 6% vs. 12% w/o preallocating...